### PR TITLE
#5 Tasks based on OOPs concepts

### PR DESCRIPTION
**Pull request by Preethxsuresh**
Requesting for review of tasks.
- TASK-1 : SHAPE HIERARCHY
- TASK-2 : EMPLOYEE HIERARCHY
- TASK-3 : BANKING SYSTEM

### CODE REVIEW

File: src/assignmentfive/BankingSystem/SavingsAccount.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SavingsAccount"/> class.
+        /// </summary>
+        /// <param name="accountNumber">Account Number.</param>
+        /// <param name="accountBalance">Account Balance.</param>
+        public SavingsAccount(string accountNumber, decimal accountBalance)
+        {
+            this.AccountNumber = accountNumber;
+            this.AccountBalance = accountBalance;
+            this.minBalance = 500;
```
Comment (nandhakumarsoliton) [inline src/assignmentfive/BankingSystem/SavingsAccount.cs:25]: This Minimum Balance - Is causing a flow error.
When I open a `Savings Account` and it asks for me to set up initial balance. There I am not getting any info like the amount should be greater than 500.
But when I try to withdraw the amount it says about Minimum Balance.

File: src/assignmentfive/ShapeHierarchy/Program.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                return;
+            }
+
+            int.TryParse(userchoice, out int choice);
+            switch (choice)
+            {
+               case 1:
+
+                    // user selected "Rectangle" option
+                    string? lengthInput = GetInput("Enter Length  : ", dimensionRegex, "Positive Numbers within 10 Digits.");
```
Comment (nandhakumarsoliton) [inline src/assignmentfive/ShapeHierarchy/Program.cs:37]: This info regarding the Valid input will be shown only when the user types an Invalid Input right?

File: src/assignmentfive/ShapeHierarchy/Rectangle.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        /// Breadth of the Rectangle.
+        /// </value>
+        public double Breadth { get; set; }
+
+        /// <summary>
+        /// Displays area and color of the Rectangle.
+        /// </summary>
+        /// <param name="color"> Color of Rectangle.</param>
+        /// <param name="area"> Area of Rectangle.</param>
+        public void PrintDetails(string color, long area)
```
Comment (nandhakumarsoliton) [inline src/assignmentfive/ShapeHierarchy/Rectangle.cs:46]: Can we remove this PrintDetails() ?

**Review by nandhakumarsoliton — CHANGES_REQUESTED** [review]

File: src/assignmentfive/ShapeHierarchy/Program.cs (hunk 2 above)
Comment (Preethxsuresh) [reply src/assignmentfive/ShapeHierarchy/Program.cs:37]: I'll add a note along when I prompt user to enter input.

**Pull request merged**